src/hooks/useExec.ts

```diff
@@ -287,6 +287,18 @@ export async function which(arg0: string | undefined) {
     return found
   }
 
+  // dark-magic will find a _lot_ of things, even when we might not want to.
+  // https://www.npmjs.com/package/sh is a great example, and probably not what
+  // we're looking for if we say `tea sh` (which is a common thing to do).
+  // So, we should only pass through if the tool we want isn't on the path.
```

Comment:
This is a good call.

We might want a whitelist.

And we may want `foo` (via magic) to behave differently to `tea foo` as well.